src/views/catalog/CreateFromInstanceTypes/hooks/useEnableInstanceTypeModal.ts

```diff
@@ -0,0 +1,41 @@
+import { useCallback } from 'react';
+
+import { INSTANCE_TYPE_ENABLED } from '@kubevirt-utils/hooks/useFeatures/constants';
+import { useFeatures } from '@kubevirt-utils/hooks/useFeatures/useFeatures';
+
+export type EnableInstanceTypeTechPreviewModalProps = {
+  canEdit: boolean;
+  isOpen: boolean;
+  onClose: () => void;
+  onEnableInstanceTypeFeature: () => void;
+};
+
+type UseEnableInstanceTypeModal = (
+  isInstanceTypeTab: boolean,
+  navigateToCatalog: () => void,
+) => EnableInstanceTypeTechPreviewModalProps & { loading: boolean };
+
+const useEnableInstanceTypeModal: UseEnableInstanceTypeModal = (
+  isInstanceTypeTab,
+  navigateToCatalog,
+) => {
+  const { canEdit, featureEnabled, loading, toggleFeature } = useFeatures(INSTANCE_TYPE_ENABLED);
+
+  const onClose = useCallback(() => {
+    navigateToCatalog();
+  }, [navigateToCatalog]);
+
+  const onEnableInstanceTypeFeature = useCallback(() => {
+    toggleFeature(true);
+  }, [toggleFeature]);
```

Comment:
i dont think usecallbak is adding any values here.. no calculation at all... and especially when line 34 isn't :) can be removed

Reply:
Actually this saves me a few not needed re-renders, since I pass these as props, the child component will re-render these callback on every child re-render. so it's true it's not crucial here, but I preffered to add this.

Reply:
>and especially when line 34 isn't :) can be removed

Do you suggest to no return these methods at all if the modal is not shown?

Reply:
No, I just think, and please test, that it will recalc when props or hook will send updates, so callbacks here don't save anything. Callbacks can save rerenders of useState changes, which affect the whole component rerender, but here u don't have any state, just passing props and hook return, so rerender will occur anyway, usecallback won't save anything but more checks :)

Reply:
It is saving a few initial renders, but it still changes on every state change of the child component so removed